src/util/time_profiler.hpp

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <array>
+#include <chrono>
+#include <numeric>
+#include <thread>
+#include <unordered_map>
+
+#include <ros/console.h>
+
+using hr_clock = std::chrono::high_resolution_clock;
+
+struct Epoch {
```

Comment:
I suppose in case I wanted to add something later, and also `durationSamples` gives a good name for what the values actually mean. I could rename to `LoopEventReadings` instead.

In fact I think I will refactor `TimeProfiler` -> `LoopProfiler` and then call epochs events. Then `reset` can be called `finishLoop` and `addEpoch` can be called `measureEvent`